clients/roscpp/src/libros/callback_queue.cpp

```diff
@@ -382,14 +393,6 @@ CallbackQueue::CallOneResult CallbackQueue::callOneCB(TLS* tls)
         result = cb->call();
       }
     }
-    catch (std::exception&)
-    {
-      // ensure that thread id gets restored, even in case of an exception
-      tls->calling_in_this_thread = last_calling;
-      throw;
```

Comment:
`throw` without an exception argument re-throws the current exception and doesn't create a new one. Therefore I don't see why this change is necessary. Can you please provide a minimal example which demonstrates why the current code is not sufficient.